src/extensions/additional/QuoteLink/QuoteLinkSpecs/md/moveLinkToQuoteAttributes.ts

```diff
@@ -0,0 +1,28 @@
+import type {PluginSimple} from 'markdown-it';
+
+export const moveLinkToQuoteAttributes: PluginSimple = (md) => {
+    md.core.ruler.push('move-link-to-quote-attributes', (state) => {
+        const {tokens} = state;
+
+        let i = 0;
+
+        while (i < tokens.length) {
+            const token = tokens[i];
+            if (token.type === 'yfm_quote-link_open') {
+                const linkToken = tokens[i + 2]?.children?.[0];
+                if (linkToken && linkToken.type === 'link_open') {
+                    token.attrSet('href', linkToken.attrGet('href') ?? '');
+                }
+
+                const linkTextToken = tokens[i + 2]?.children?.[1];
+                if (linkTextToken?.content) {
+                    token.attrSet('content', linkTextToken.content);
+                }
+
+                tokens.splice(i + 1, 3);
```

Comment:
should splice `tokens[i+1].children` – children of inline token

but first you need to find the index of the `link_close` token, and remove tokens starting `link_open` and ending `link_close`